### PR TITLE
Explicitly disallow tainting controller nodes

### LINKED ISSUE #754: Disallow specifying `experimental.taints`
The `experimental.taints` key had been added to cluster.yaml via #132 in the end of last year for tainting controller nodes.
As the name doesn't express the fact that it is only for controller nodes very well, and also it has been several releases since it is added as an experimental feature, I suggest renaming.

cc @redbaron @c-knowles @danielfm @camilb

### PR DESCRIPTION
**Pull request by mumoshu**
This PR depends on #755. Please see the last commit for changes made solely in this PR.

Resolves #754

This is verified manually by running E2E

### CODE REVIEW

File: core/nodepool/config/config.go [hunk 1]
```diff
@@ -332,6 +332,10 @@ func (c ProvidedConfig) valid() error {
 		return err
 	}
 
+	if err := c.NodeSettings.Validate(); err != nil {
```
Comment (danielfm) [inline core/nodepool/config/config.go]: nit for a later PR: standardize the name of these validation functions.

Comment (mumoshu) [inline core/nodepool/config/config.go]: Must do! Submitted the issue #759 to track it.

**Comment by mumoshu** [comment]
@danielfm Thanks for your review 👍

**Pull request merged**